src/methods/join.js

```diff
@@ -186,7 +217,7 @@ export function joinEffectsAndRemoveNestedReturnCompletions(
       e1[0] = new ReturnCompletion(realm.intrinsics.undefined);
       return joinEffects(realm, c.joinCondition, e1, e2);
     } else {
-      throw AbstractValue.createIntrospectionErrorThrowCompletion(c.joinCondition);
+      throw c;
```

Comment:
Some callers can now deal with PossiblyNormalCompletions from calls, so move the error upstream.